src/frontend/src/lib/components/address-book/AddressBookModal.svelte

```diff
@@ -153,11 +158,25 @@
 	testId={ADDRESS_BOOK_MODAL}
 	on:nnsClose={close}
 >
-	<svelte:fragment slot="title"
-		>{currentStepName === AddressBookSteps.EDIT_CONTACT_NAME && nonNullish(editContactNameStep)
-			? editContactNameStep.title
-			: (currentStep?.title ?? '')}</svelte:fragment
-	>
+	<svelte:fragment slot="title">
+		{#if currentStepName === AddressBookSteps.SHOW_ADDRESS}
+			<div class="flex flex-wrap items-center gap-2">
+				<Avatar
+					name={currentContact?.name}
+					variant="xs"
+					styleClass="rounded-full flex items-center justify-center"
+				/>
+				<div class="text-center text-lg font-semibold text-primary">
+					{currentContact?.name}
+				</div>
+			</div>
+		{:else if currentStepName === AddressBookSteps.EDIT_CONTACT_NAME && nonNullish(editContactNameStep)}
+			{editContactNameStep.title}
```

Comment:
to simplify, i would do

```suggestion
		{:else if currentStep?.name === AddressBookSteps.EDIT_CONTACT_NAME}
			{currentStep.title}
```